distributor/src/distributor/filters.bal

```diff
@@ -4,26 +4,59 @@ import ballerina/http;
 import ballerina/log;
 import ballerina/websub;
 
+const WWW_AUTHENTICATE_HEADER = "WWW-Authenticate";
+
 const HUB_TOPIC = "hub.topic";
 const HUB_CALLBACK = "hub.callback";
 
-map<string> callbackMap = {};
+map<string> resultCallbackMap = {};
+map<string> imageCallbackMap = {};
+
+# Filter to challenge authentication.
+public type AuthChallengeFilter object {
+    *http:RequestFilter;
+
+    public function filterRequest(http:Caller caller, http:Request request, http:FilterContext context) 
+                        returns boolean {
+        if request.hasHeader(http:AUTH_HEADER) || request.rawPath != "/" {
```

Comment:
Ideally we need to tie this to the auth session model that the basic auth handler implements. The check basically is "are you already authenticated?" and if not, if you haven't given the Authorization: header then we need to challenge.

I'll go ahead with this for now - its not perfect but its safe as the basic auth handler will fail if it isn't satisfied.